R/decisiontreeutils.R

```diff
@@ -0,0 +1,431 @@
+    )
+  } else {
+    stop(paste("Unsupported function specified: ", f_string))
+  }
+}
+
+#' adjusts config based on results if config was initially "Auto"
+#'
+#' @param config list of config options
+#' @param the_model model object
+#' @return model obj after adjusting complexity parameter
+adjustCP <- function(config, the_model) {
+  if(is.na(as.numeric(config$the.cp)) && (config$the.cp == "Auto" || config$the.cp == "")) {
+    cp_table <- as.data.frame(the_model$cptable)
+    pos_cp <- cp_table$CP[(cp_table$xerror - 0.5*cp_table$xstd) <= min(cp_table$xerror)]
+    new_cp <- pos_cp[1]
+    print(cp_table)
+    if (cp_table$xerror[1] == min(cp_table$xerror)) {
+      stop.Alteryx("The minimum cross validation error occurs for a CP value where there are no splits. Specify a complexity parameter and try again.")
+    }
+    prune(the_model, cp = new_cp)
+  } else {
+    the_model
+  }
+}
+
+#' get grp|out pipes for outputting static report
+#'
+#' @param config list of config options
+#' @param the_model model object
+#' @param is_XDF boolean of whether model is XDF
+#' @return dataframe of piped results
+getDTPipes <- function(config, the_model, is_XDF) {
+
+  # The output: Start with the pruning table (have rxDTree objects add rpart
+  # inheritance for printing and plotting purposes).
+  if (is_XDF) {
+    the_model_rpart <- rxAddInheritance(the_model)
+    printcp(the_model_rpart)
+    out <- capture.output(printcp(the_model_rpart))
+    the_model$xlevels <- do.call(match.fun("xdfLevels"), list(paste0("~ ", paste(names_x_vars, collapse = " + ")), xdf_path))
+    if (is.factor(the_target)) {
+      target_info <- do.call(match.fun("rxSummary"), list(paste0("~ ", name_y_var), data = xdf.path))[["categorical"]]
+      if(length(target_info) == 1) {
+        the_model$yinfo <- list(levels = as.character(target_info[[1]][,1]), counts = target_info[[1]][,2])
+      }
+    }
+  } else {
+    printcp(the_model) # Pruning Table
+    out <- capture.output(printcp(the_model))
+  }
+
+  out <- out[out != ""]
+  print(out)
+  out <- out[2:length(out)]
+  # Find the index of the end of the call
+  for (i in 1:length(out)) {
+    if (grepl("Variables", out[i])) {
+      end_call <- i - 1
+      break
+    }
+  }
+  call1 <- out[1:end_call]
+  the_call = gsub("\\s\\s", "", paste(call1, collapse=""))
+  # Figure out where the summary information ends and the actual pruning table
+  # starts. The header for the pruning table is removed, and will instead show-up
+  # in the Alteryx reporting tool
+  for (i in 1:length(out)) {
+    if (grepl("\\sCP\\s", out[i])) {
+      strt_table <- i + 1
+      break
+    }
+  }
+  model_sum <- length((end_call + 1):(strt_table - 2))
+  prune_tbl1 <- out[strt_table:length(out)]
+  out <- c(the_call, out[(end_call + 1):(strt_table - 2)])
+  rpart_out <- data.frame(grp = c("Call", rep("Model_Sum", model_sum)), out = out)
+  rpart_out$grp <- as.character(rpart_out$grp)
+  rpart_out$out <- as.character(out)
```

Comment:
Similar to my comments on the pruning table, the pipe syntax can simplify lines 186 - 219 as well

```r
out <-  capture.output(printcp(the_model))
model_sum = out %>%
  extract(1:grep("^n=", .)) %>%
  .[. != ""] %>%
  data.frame(grp = "Model_Sum", out = ., stringsAsFactors = FALSE)

the_call <- out %>%
  extract(2:(grep("^Variables", .) - 1)) %>%
  .[. != ""] %>%
  paste(collapse = "") %>%
  data.frame(grp = "Call", out = ., stringsAsFactors = FALSE)
```